### PR TITLE
Tank IFF, Minigun tweaks, code cleanup & new hitsound

### PR DESCRIPTION
**Pull request by TheManWithNoHands**
# About the pull request

* Adds IFF to the Minigun
* Changes used projectile icon for Minigun
* Adds scatter decrease to Minigun with windup
* increases firing speed of the minigun
* increases rev up time of the minigun
* Puts Minigun bullet code into the tank ammo file, for organisation.
* Adds hitsound for Autocannon Flakbullets
* Adds new projectile icon for Autocannon
* Fixes code for UPP minigun ammo

# Explain why it's good for the game

The Minigun for the Tank is lacking IFF for no good reason. Its a left over from old code and isnt adjusted for its current state.
This should make the Minigun less ass to use, and also make the Minigun a better choice overall.
The autocannon keeps FF, to have to role of a more high risk/high damage weapon. Its semi automatic, so it can affort not getting IFF.

The Minigun also gets thigther spread as it spins up. It starts at a higher spread in turn.
The full time to achive full spin has also been increased by 2 seconds, and the overall firingspeed has been very very sligthly increased.
This overall should make the minigun a very good support weapon. Baiscly the SGs bigger cousin, just that the gun is actually fun. I hope. to eleate it being the SGs bigger cousin, the gun now no longer has smaller projectile sprites then the SG. Autocannon also gets new sprites fitting for its 20mm rounds.
These changes should elevate the minigun to become a more stable pick and less of a noob trap.

The new hitsound is sampeld straigth from pixabay or our repo. So no legal troubles there.

Thanks to cuberound that helped with the minigun scatter code

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>
grafik

</details>

# Changelog
:cl: TheManWithNoHands
balance: IFF, scatter based on minigun rev up, rev up time increase, firing delay decrease
code: Places minigun aMmo in the Tank ammo File
soundadd: adds a hit sound for 20mm flak
/:cl:

### CODE REVIEW

**Comment by Drulikar** [comment]
stale exempt until reviewed

File: code/datums/ammo/bullet/tank.dm [hunk 1]
```diff
@@ -72,3 +77,32 @@
...
+	penetration = ARMOR_PENETRATION_TIER_6
+
+/datum/ammo/bullet/tank/minigun/New()
+	..()
+	if(SSticker.mode && MODE_HAS_FLAG(MODE_FACTION_CLASH))
+		damage = 15
+	else if(SSticker.current_state < GAME_STATE_PLAYING)
+		RegisterSignal(SSdcs, COMSIG_GLOB_MODE_PRESETUP, PROC_REF(setup_hvh_damage))
+
+/datum/ammo/bullet/tank/minigun/proc/setup_hvh_damage()
```
Comment (kiVts) [inline code/datums/ammo/bullet/tank.dm:101]: Every proc that is registered in register signal must have SIGNAL_HANDLER on the top.

File: code/datums/ammo/bullet/tank.dm [hunk 2]
```diff
@@ -4,9 +4,14 @@
...
 */
 
+//To do: Move the rest of the Tank ammo into here so its all organized. They are currently all over the place.
+
+//Autocannon Ammo//
+
 /datum/ammo/bullet/tank/flak
 	name = "flak autocannon bullet"
 	icon_state = "autocannon"
+	sound_hit  = list('sound/weapons/sting_boom_small1.ogg')
```
Comment (kiVts) [inline code/datums/ammo/bullet/tank.dm:14]: add the path to file in get_sfx and link it here instead, look at how other sound_hit is done, this is mostly for consistency though.

**Review by kiVts — DISMISSED** [review]

**Comment by TheManWithNoHands** [comment]
How the hell did i start a review by answering someone.

**Review by Blundir — APPROVED** [review]

File: code/datums/ammo/bullet/tank.dm [hunk 3]
```diff
@@ -4,9 +4,14 @@
 //======
 */
 
+//To do: Move the rest of the Tank ammo into here so its all organized. They are currently all over the place.
+
+//Autocannon Ammo//
```
Comment (harryob) [inline code/datums/ammo/bullet/tank.dm:9]:
```suggestion
//Autocannon Ammo//
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**